crates/flowistry_pdg_construction/src/construct.rs

```diff
@@ -655,6 +668,57 @@ impl<'tcx> GraphConstructor<'tcx> {
         self.tcx.def_path_str(def_id)
     }
 
+    /// Special case behavior for calls to functions used in desugaring async functions.
+    ///
+    /// Ensures that functions like `Pin::new_unchecked` are not modularly-approximated.
+    fn approximate_async_functions(
+        &self,
+        state: &mut PartialGraph<'tcx>,
+        def_id: DefId,
+        args: &[Operand<'tcx>],
+        destination: Place<'tcx>,
+        location: Location,
+    ) -> bool {
+        let lang_items = self.tcx.lang_items();
+        if Some(def_id) == lang_items.new_unchecked_fn() {
+            let [op] = args else {
+                unreachable!();
+            };
+            let mut operands = IndexVec::new();
+            operands.push(op.clone());
+            let TyKind::Adt(adt_id, generics) =
+                destination.ty(self.body.as_ref(), self.tcx).ty.kind()
+            else {
+                unreachable!()
+            };
+            assert_eq!(adt_id.did(), lang_items.pin_type().unwrap());
+            let aggregate_kind =
+                AggregateKind::Adt(adt_id.did(), VariantIdx::from_u32(0), generics, None, None);
+            let rvalue = Rvalue::Aggregate(Box::new(aggregate_kind), operands);
+            trace!("Handling new_unchecked as assign for {destination:?}");
+            self.modular_mutation_visitor(state)
+                .visit_assign(&destination, &rvalue, location);
+            true
+        } else if Some(def_id) == lang_items.into_future_fn()
+            // FIXME: better way to get retrieve this stdlib DefId?
+            || self.tcx.def_path_str(def_id) == "<F as std::future::IntoFuture>::into_future"
```

Comment:
I think there are two ways. Either pass in the unresolved `DefId`, which should be the one for the trait item, or use something like this 

```rs
(tcx.def_kind(def_id) == DefKind::AssocFn).then(||
    tcx.associated_item(def_id).trait_item_def_id
).flatten() == lang_items.into_future_fn()
```

But I can add that later